### PR TITLE
Refactor Makefile docker variant targets and fix for m1 machines

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

- Refactor the Makefile grouping targets based on variants to simplify large make targets
- Adds the amd64 platform for all distroless base images since we only publish amd64 images and use amd64 envoy. This also unblocks distroless dev on m1. 
When building distroless images on m1, it fails when using the distroless-base image even if it exists as the arch is amd64. It tries to fetch the arm64 distroless-base image from quay that doesn't exist and shows this error
`
ERROR: failed to solve: [quay.io/solo-io/distroless-base:1.0.1-dev](http://quay.io/solo-io/distroless-base:1.0.1-dev): failed to resolve source metadata for [quay.io/solo-io/distroless-base:1.0.1-dev](http://quay.io/solo-io/distroless-base:1.0.1-dev): unexpected status from HEAD request to https://quay.io/v2/solo-io/distroless-base/manifests/1.0.1-dev: 401 UNAUTHORIZED
`
This fixes it by ensuring we only use amd64 images for distroless builds

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -867,82 +867,102 @@ docker-push-%-distroless:
...
+.PHONY: docker-distroless-retag
+docker-distroless-retag: docker-retag-gloo
+docker-distroless-retag: docker-retag-discovery
+docker-distroless-retag: docker-retag-gloo-envoy-wrapper
+docker-distroless-retag: docker-retag-sds
+docker-distroless-retag: docker-retag-certgen
+docker-distroless-retag: docker-retag-ingress
+docker-distroless-retag: docker-retag-access-logger
+docker-distroless-retag: docker-retag-kubectl
+
```
Comment (sheidkamp) [inline Makefile:955]: Should these be `-distroless`?

**Review by sheidkamp — CHANGES_REQUESTED** [review]
One spot where it looks like there was a c/p error. Otherwise LGTM.

File: Makefile [hunk 2]
```diff
@@ -867,82 +867,102 @@ docker-push-%-distroless:
...
+.PHONY: docker-distroless
+docker-standard: check-go-version
+docker-standard: gloo-distroless-docker
+docker-standard: discovery-distroless-docker
+docker-standard: gloo-envoy-wrapper-distroless-docker
+docker-standard: sds-distroless-docker
+docker-standard: certgen-distroless-docker
+docker-standard: ingress-distroless-docker
+docker-standard: access-logger-distroless-docker
+docker-standard: kubectl-distroless-docker
```
Comment (sheidkamp) [inline Makefile:890]: these `docker-standards` should be `docker-distroless`

**Review by sheidkamp — CHANGES_REQUESTED** [review]
one more c/p error

**Review by sheidkamp — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (davidjumani) [reply Makefile:955]: Good catch! Thanks

File: Makefile (hunk 2 above)
Comment (davidjumani) [reply Makefile]: Good catch! Thanks

File: ci/kind/setup-kind.sh [hunk 3]
```diff
@@ -18,6 +18,8 @@ KUBE2E_TESTS="${KUBE2E_TESTS:-gateway}"  # If 'KUBE2E_TESTS' not set or null, us
 # The version of istio to install for glooctl tests
 # https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases
 ISTIO_VERSION="${ISTIO_VERSION:-1.18.2}"
+# Set the default image variant to standard
+IMAGE_VARIANT="${IMAGE_VARIANT:-standard}"
```
Comment (bewebi) [inline ci/kind/setup-kind.sh:22]: Doesn't this need to get passed as an env var to step #2 below?

**Review by sheidkamp — APPROVED** [review]

**Review by bewebi — DISMISSED** [review]
LGTM

**Review by nfuden — APPROVED** [review]

**Review by sheidkamp — APPROVED** [review]

**Pull request merged**